lib/supavisor/protocol/server.ex

```diff
@@ -360,18 +361,13 @@ defmodule Supavisor.Protocol.Server do
     [<<?S, len::integer-32>>, payload]
   end
 
-  @spec backend_key_data() :: iodata()
+  @spec backend_key_data() :: {iodata(), binary}
   def backend_key_data() do
-    procid = System.unique_integer([:positive, :monotonic])
-    secret = Enum.random(0..9_999_999_999)
-    payload = <<procid::integer-32, secret::integer-32>>
+    pid = System.unique_integer([:positive, :monotonic])
+    key = Enum.random(0..9_999_999_999)
```

Comment:
This may be a better generator:

```suggestion
    key = :crypto.strong_rand_bytes(4)
```